tests/tr1/tests/functional9/test.cpp

```diff
@@ -60,6 +60,19 @@ static void t_invoke() { // test invoke
     CHECK_INT(STD invoke(sum4, 1, 3, 5, 7), 16);
 }
 
+// COMPILE-ONLY
+void test_LWG_3146() {
+    int i = 0;
+    STD reference_wrapper<int> ri(i);
+    STD reference_wrapper<STD reference_wrapper<int>> rri(ri);

```

Comment:
Why the nested type

Reply:
LWG has the example which shows what is broken with current implementation and why the change is needed.
```
int i = 0;
std::reference_wrapper<int> ri(i);
std::reference_wrapper<std::reference_wrapper<int>> rri(ri);
std::ref(rri); // error within 'std::ref'
```